client/packages/requisitions/src/RequestRequisition/DetailView/ContentArea.tsx

```diff
@@ -8,40 +8,23 @@ import {
 } from '@openmsupply-client/common';
 import { RequestLineFragment, useHideOverStocked, useRequest } from '../api';
 import { isRequestLinePlaceholderRow } from '../../utils';
-import { RequestItem } from '../../types';
 
 interface ContentAreaProps {
   onAddItem: () => void;
   onRowClick: null | ((line: RequestLineFragment) => void);
 }
 
 const useHighlightPlaceholderRows = (
-  rows: RequestLineFragment[] | RequestItem[] | undefined
+  rows: RequestLineFragment[] | undefined
 ) => {
   const { setRowStyles } = useRowStyle();
 
   useEffect(() => {
     if (!rows) return;
-    const placeholders = [];
-
-    for (const row of rows) {
-      if ('requestedQuantity' in row) {
-        if (isRequestLinePlaceholderRow(row)) {
-          placeholders.push(row.id);
-        }
-      } else {
-        const hasPlaceholder = row.lines.some(isRequestLinePlaceholderRow);
-        if (hasPlaceholder) {
-          row.lines.forEach(line => {
-            if (isRequestLinePlaceholderRow(line)) {
-              placeholders.push(line.id);
-            }
-          });
-          placeholders.push(row.id);
-        }
-      }
-    }
 
+    const placeholders = rows
+      .filter(isRequestLinePlaceholderRow)
+      .map((row: { id: any; }) => row.id);
```

Comment:
With the change to the type of `row` to  an array of `RequestLineFragment`, this simplification is now type safe 😄 
```suggestion
      .map(row => row.id);
```